Source/DafnyCore/Backends/Dafny/DafnyCodeGenerator.cs

```diff
@@ -321,28 +362,33 @@ class DafnyCodeGenerator : SinglePassCodeGenerator {
 
         List<DAST.Statement> witnessStmts = new();
         DAST.Expression witness = null;
+        var statementBuf = new StatementBuffer();
```

Comment:
StatementBuffer sounds like it might solve the same problem as the `wStmts` parameter that the SinglePassCompiler passes around, and might just need to be added to this method too. I could consult with @fabiomadge.

Reply:
`statementBuf` is actually used to create a wStmts via `new BuilderSyntaxTree<StatementContainer>(statementBuf, this)`. Since this method encodes a declaration, there is at this point no possibility to have a wStmt, so it has to be created.